src/main/java/com/google/firebase/auth/internal/BaseCredential.java

```diff
@@ -59,14 +60,14 @@ public final GoogleCredentials getGoogleCredentials() {
 
   @Override
   public Task<GoogleOAuthAccessToken> getAccessToken() {
-    return Tasks.call(new Callable<GoogleOAuthAccessToken>() {
-      @Override
-      public GoogleOAuthAccessToken call() throws Exception {
-        AccessToken accessToken = googleCredentials.refreshAccessToken();
-        return new GoogleOAuthAccessToken(accessToken.getTokenValue(),
-            accessToken.getExpirationTime().getTime());
-      }
-    });
+    try {
+      AccessToken accessToken = googleCredentials.refreshAccessToken();
+      GoogleOAuthAccessToken googleToken = new GoogleOAuthAccessToken(accessToken.getTokenValue(),
+          accessToken.getExpirationTime().getTime());
+      return Tasks.forResult(googleToken);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
```

Comment:
Does the Task library not have its own way of passing along errors?

Reply:
Done. On a related note that both `Task` and `FirebaseCredential` are now deprecated, and are on their way out.